contracts/PBAB+Collabs/GenArt721MinterDAExp_PBAB.sol

```diff
@@ -0,0 +1,584 @@
+// SPDX-License-Identifier: LGPL-3.0-only
+// Created By: Art Blocks Inc.
+
+import "../interfaces/0.8.x/IGenArt721CoreV2_PBAB.sol";
+
+import "@openzeppelin-4.5/contracts/security/ReentrancyGuard.sol";
+import "@openzeppelin-4.5/contracts/utils/math/SafeCast.sol";
+
+pragma solidity 0.8.17;
+
+/**
+ * @title A minter contract that allows tokens to be minted with ETH.
+ * Pricing is achieved using an automated Dutch-auction mechanism.
+ * This is a fork of MinterDAExpV2, which is intended to be used directly
+ * with IGenArt721CoreV2_PBAB contracts rather than with IGenArt721CoreContractV3
+ * and as such does not conform to IFilteredMinterV0 nor assume presence
+ * of a IMinterFilterV0 conforming minter filter.
+ * @author Art Blocks Inc.
+ * @notice Privileged Roles and Ownership:
+ * This contract is designed to be managed, with limited powers.
+ * Privileged roles and abilities are controlled by the core contract's Admin
+ * ACL contract and a project's artist. Both of these roles hold extensive
+ * power and can modify minter details.
+ * Care must be taken to ensure that the admin ACL contract and artist
+ * addresses are secure behind a multi-sig or other access control mechanism.
+ * ----------------------------------------------------------------------------
+ * The following functions are restricted to the core contract's Admin ACL
+ * contract:
+ * - setAllowablePriceDecayHalfLifeRangeSeconds (note: this range is only
+ *   enforced when creating new auctions)
+ * - resetAuctionDetails (note: this will prevent minting until a new auction
+ *   is created)
+ * ----------------------------------------------------------------------------
+ * The following functions are restricted to a project's artist:
+ * - setAuctionDetails (note: this may only be called when there is no active
+ *   auction)
+ * ----------------------------------------------------------------------------
+ * Additional admin and artist privileged roles may be described on other
+ * contracts that this minter integrates with.
+ *
+ * @dev Note that while this minter makes use of `block.timestamp` and it is
+ * technically possible that this value is manipulated by block producers, such
+ * manipulation will not have material impact on the price values of this minter
+ * given the business practices for how pricing is congfigured for this minter
+ * and that variations on the order of less than a minute should not
+ * meaningfully impact price given the minimum allowable price decay rate that
+ * this minter intends to support.
+ */
+contract GenArt721MinterDAExp_PBAB is ReentrancyGuard {
```

Comment:
WDYT about calling this `MinterDAExpEngineV0`? - The GenArt721 prefix is something we dropped off of our minter suite to simplify the naming, and I think probably makes sense to do the same here (also applies to file name and `minterType` constant below)

Reply:
My inclination would be to keep the file naming the same as it is now in the PR, just for consistency with the other 2 V2-Engine compatible minters, but it SGTM to update the `minterType`. 

That said, thinking about it more, I say we just remove the `minterType` altogether actually, as we don't intend for this minter to be mintersuite compatible anyway.

<img width="364" alt="Screenshot 2023-01-12 at 5 46 18 PM" src="https://user-images.githubusercontent.com/8602661/212211373-75831130-92be-4020-b0bd-a256fe28cbbd.png">

Given that we're going to be writing a new set of minters to be compatible with a V3-engine-plus-minter-suite world, I think backwards consistency for now in these V2-compatible minters makes sense (as we'll be archiving them all anyway by the EOQ).

WDYT ser? @ryley-o

Reply:
yep that makes sense ser 👍 🙏

Reply:
Rad–thank you for confirming ser. :)